File: pset6/readability/readability.py
import cs50


def main():
    ##Input String or Text
    text = str(input("Text: "))
    A = characters(text)
    B = words(text)
    C = sentences(text)
    ##print(A, B, C)

    avg_letters = float(A/B*100)
    avg_sentences = float(C/B*100)
    index = float(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)
    rounded_index = round(index)

    if (index < 1):
        print("Before Grade 1");
    elif (index > 16):
        print("Grade 16+");
    else:
        print("Grade: ", rounded_index);



##Count number of characters in string
def characters(text):
    counter_characters = 0

    for i in range(len(text)):
        if (text[i].isalpha() == True):
            counter_characters +=1
    ##print("Number of characters:", counter_characters)
    return counter_characters



##Count number of words

def words(text):
    counter_words = 1
    for i in range(len(text)):
        if (text[i] == " "):
            counter_words += 1
    ##print("Number of words:", counter_words)
    return counter_words

##Count number of sentences
def sentences(text):
    counter_sentences = 0
    for i in range(len(text)):
        if (text[i] == '.' or text[i] == '!' or text[i] == '?' ):
            counter_sentences += 1
    ##print("Number of sentences:", counter_sentences)
    return counter_sentences


main()

File: Misc/readability.c
//includes
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>


//functions
int letters(string a);
int words(string b);
int sentences(string c);

int main(void)
{
    //something to get the string from the user
    string text = get_string("What's the text we're evaluating?: ");

    //count within the string the letters. I need to cycle through the array and add 1 everytime it accounters a letter or a symbol
    int number_letters = letters(text);
    //printf("%i letter(s)\n", number_letters);

    //count within the string the words. I maybe could get away with counting spaces here
    int number_words = words(text);
    //printf("%i word(s)\n", number_words);

    //count within the string how many setences. I need to find the . or !
    int number_sentences = sentences(text);
    //printf("%i sentence(s)\n", number_sentences);

    //Return the index
    float L = (float)number_letters / number_words  * 100;
    float S = (float)number_sentences / number_words * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int rounded_index = round(index);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", rounded_index);
    }



}

//Function of Letters

int letters(string a)
{
    int letters = 0;
    for (int i = 0; a[i] != '\0'; i++)
    {
        if (isalpha(a[i]))
        {
            letters = letters + 1;
            //printf("Character ASCII %i i Value %i Letters %i\n", a[i], i, letters);
        }

    }
    return letters;
}

//Function of Words

int words(string b)
{
    int words = 1;
    for (int i = 0; b[i] != '\0'; i++)
    {
        if (isspace(b[i]))
        {
            words = words + 1;
               //printf("Value %i number of words %i\n", i, words);
        }
    }
    return words;
}

//Function of Sentences

int sentences(string c)
{
    int sentences = 0;
    for (int i = 0; c[i] != '\0'; i++)
    {
        if (c[i] == '.' || c[i] == '!' || c[i] == '?' )
        {
            sentences = sentences + 1;
            //printf("Value %i numner of sentences %i\n", i, sentences);
        }
    }
    return sentences;
}